src/main/kotlin/no/skatteetaten/aurora/gobo/integration/dbh/databaseServices.kt

```diff
@@ -38,77 +38,86 @@ class DatabaseServiceReactive(
         .uri("/api/v1/admin/databaseInstance/")
-        .authHeader()
-        .retrieve()
-        .bodyToMono<DbhResponse<*>>()
-        .items()
+        .retrieveAuthenticatedItems()
 
     fun getDatabaseSchemas(affiliation: String): Mono<List<DatabaseSchemaResource>> = webClient
         .get()
         .uri {
             it.path("/api/v1/schema/").queryParam("labels", "affiliation=$affiliation").build()
         }
-        .authHeader()
-        .retrieve()
-        .bodyToMono<DbhResponse<*>>()
-        .items()
+        .retrieveAuthenticatedItems()
 
     fun getDatabaseSchema(id: String): Mono<DatabaseSchemaResource> = webClient
         .get()
         .uri("/api/v1/schema/$id")
-        .authHeader()
-        .retrieve()
-        .bodyToMono<DbhResponse<*>>()
-        .item()
+        .retrieveAuthenticatedItem()
+
+    fun getRestorableDatabaseSchemas(affiliation: String): Mono<List<RestorableDatabaseSchemaResource>> = webClient
+        .get()
+        .uri {
+            it.path("/api/v1/restorableSchema/").queryParam("labels", "affiliation=$affiliation").build()
+        }
+        .retrieveAuthenticatedItems()
 
     fun updateDatabaseSchema(input: SchemaUpdateRequest): Mono<DatabaseSchemaResource> = webClient
         .put()
         .uri("/api/v1/schema/${input.id}")
         .body(BodyInserters.fromValue(input))
-        .authHeader()
-        .retrieve()
-        .bodyToMono<DbhResponse<*>>()
-        .item()
+        .retrieveAuthenticatedItem()
 
-    fun deleteDatabaseSchemas(input: List<SchemaDeletionRequest>): Flux<SchemaDeletionResponse> {
+    fun deleteDatabaseSchemas(input: List<SchemaDeletionRequest>): Flux<SchemaCooldownChangeResponse> {
         val responses = input.map { request ->
             val requestSpec = webClient
                 .delete()
                 .uri("/api/v1/schema/${request.id}")
-                .authHeader()
 
             request.cooldownDurationHours?.let {
                 requestSpec.header(HEADER_COOLDOWN_DURATION_HOURS, it.toString())
             }
 
-            requestSpec.retrieve().bodyToMono<DbhResponse<*>>().map {
+            requestSpec.retrieveAuthenticated().bodyToDbhResponse().map {
                 request.id to it
             }
         }
 
-        return Flux.merge(responses).map { SchemaDeletionResponse(id = it.first, success = it.second.isOk()) }
+        return Flux.merge(responses).map { SchemaCooldownChangeResponse(id = it.first, success = it.second.isOk()) }
     }
 
-    fun testJdbcConnection(id: String? = null, user: JdbcUser? = null): Mono<ConnectionVerificationResponse> {
-        val response: Mono<DbhResponse<ConnectionVerificationResponse>> = webClient
-                .put()
-                .uri("/api/v1/schema/validate")
+    fun restoreDatabaseSchemas(input: List<SchemaRestorationRequest>): Flux<SchemaCooldownChangeResponse> {
+        val responses = input.map { request ->
+            val requestSpec = webClient
+                .patch()
```

Comment:
Er ikke helt sikker på om jeg er enige i valget av metode og url til dette endepunktet. Vi kaller en patch til et endepunkt som er veldig spesifikt for å restore et skjema. Kan man sende noe annet til dette endepunktet en akkurat dette requestet?

Reply:
Det finnes et annet endepunkt uten ID'en, denne har HTTP GET. eksempel:
GET /api/v1/restorableSchema/

Hva føler du er galt med en PATCH? 
Er det ressursen RestorabledSchema du ikke er enig i?

Reply:
PATCH brukes vel gjerne hvis man skal sende inn deler av en ressurs og skal oppdatere det som ligger der i henhold til dette. Og slik jeg leser koden her er det ingen andre handlinger på restorable schema som er POST eller PUT.